src/components/layout/Layout.tsx

```diff
@@ -0,0 +1,115 @@
+import { Box, Typography, useMediaQuery, useTheme } from '@mui/material';
+import { Link, Outlet, useLocation,useNavigate } from 'react-router-dom';
+import Sidebar from './Sidebar';
+import { Menu, SettingsOutlined } from '@mui/icons-material';
+import RowContainerBetween from '../shared/RowContainerBetween';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import { DevicesContext } from '../../context/devices.context';
+function Layout() {
+    const {setAccessToken, setProfile} = useContext(DevicesContext);
+    const navigate = useNavigate();
+    const reToken =useCallback(() => {
+        window.wazigate.set<string>("auth/retoken", {})
+        .then(async(res)=>{
+            setAccessToken(res);
+            await window.wazigate.setToken(res);
+        })
+        .catch((error) => {
+            console.log(error);
+        });
+    },[setAccessToken]); 
+    const isAuthorized =useCallback(() => {
+        fetch("sys/uptime")
+        .then((resp)=>{
+            if(resp.status === 401){
+                const creds = window.sessionStorage.getItem("creds");
+                if(creds){
+                    const {username,password} = JSON.parse(creds);
+                    window.wazigate.authToken(username,password)
+                    .then((res)=>{
+                        setAccessToken(res)
+                        window.wazigate.setToken(res);
+                    })
+                    .catch(()=>{
+                        setAccessToken('');
+                        setProfile(null);
+                        navigate('/')
+                    })
+                }else{
+                    setAccessToken('');
+                    setProfile(null);
+                    navigate('/');
+                }
+            }  
+        })
+        .catch(()=>{
+            (document.getElementById("dashboard") as HTMLElement).innerHTML = "<div style='margin-top: 10%;color:black; text-align: center;border: 1px solid #BBB;border-radius: 5px;padding: 5%;margin-left: 10%;margin-right: 10%;background-color: #EEE;'><h1>Wazigate is not accessible...</h1></div>";
+        });
+    },[navigate, setProfile, setAccessToken]);
+    useEffect(()=>{
+        const int = setInterval(reToken, 1000 * 60 * 5);
+        const timer = setInterval(isAuthorized, 1000 * 15);
+        return ()=>{
```

Comment:
It's not clear why you need to use "isAuthorized" if you already have "reToken".

Reply:
It's also set this way in the previous dashboard codebase, so probably @j-forster  can comment on this.